web_exportall/controllers/main.py

```diff
@@ -0,0 +1,201 @@
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import csv
+import operator
+import simplejson
+import re
+from cStringIO import StringIO
+try:
+    import xlwt
+except ImportError:
+    xlwt = None
+
+import openerp
+from openerp import sql_db
+from openerp.addons.web.controllers.main import (ExportFormat,
+                                                 serialize_exception)
+from openerp import http
+from openerp.http import request
+
+
+def itter_ids(all_ids, step_size=250):
+    """ Splits all_ids into more manageable chunks of
+        size step_size and yields them """
+    offset = 0
+    while True:
+        id_set = all_ids[offset:offset+step_size]
+        offset += step_size
+        if len(id_set):
+            yield id_set
+        else:
+            return
+
+
+def itter_data(Model, dbname, uid, all_ids, field_names, raw_data, context):
+    """ Reads data in chunks of ids from all_ids and
+        yields the rows one by one """
+    cr = sql_db.db_connect(dbname).cursor()
+    with cr:
+        with openerp.api.Environment.manage():
+            for ids in all_ids:
+                import_data = Model.export_data(cr,
+                                                uid,
+                                                ids,
+                                                field_names,
+                                                raw_data,
+                                                context=context
+                                                ).get('datas', [])
+                for data in import_data:
+                    yield data
+    return
+
+
+class ExportAllBase(ExportFormat):
+    raw_data = False
+
+    def base(self, data, token):
+        data_dict = simplejson.loads(data)
+        model = data_dict.get('model', False)
+        fields = data_dict.get('fields', [])
+        ids = data_dict.get('ids', [])
+        domain = data_dict.get('domain', False)
+        import_compat = data_dict.get('import_compat', False)
+        context = data_dict.get('context', request.context)
+
+        field_names = map(operator.itemgetter('name'), fields)
+
+        Model = request.registry.get(model)
+        ids = itter_ids(Model.search(request.cr,
+                                     request.uid,
+                                     domain,
+                                     context=context))
+        import_data = itter_data(Model,
+                                 request.cr.dbname,
+                                 request.uid,
+                                 ids,
+                                 field_names,
+                                 self.raw_data,
+                                 context)
```

Comment:
You can save a lot of lines by making your line break right after `(` and it keeps the code from looking lopsided.
```python
        import_data = itter_data(
            Model, request.cr.dbname, request.uid, ids, field_names,
            self.raw_data, context
        )
```